sync/task_pool.go

```diff
@@ -158,14 +159,54 @@ func (p *TaskPool) run(id int, q chan task) error {
 	}
 }
 
-// add task
+// AddTask wait idle worker add task
 func (p *TaskPool) AddTask(t task) {
+	idx := atomic.AddUint32(&p.idx, 1)
+	id := idx % uint32(p.tQNumber)
+
+	select {
+	case <-p.done:
+		return
+	case p.qArray[id] <- t:
```

Comment:
I think `AddTask` should make sure task being added and executed, so change it to :
```go
select {
	case <-p.done:
		return
       default:
                p.qArray[id] <- t
}
```

Reply:
if p.done is ok, both `case` and `default` are not work.

I add a return value means success status, both them could get a false value